programs/stake/src/stake_instruction.rs

```diff
@@ -5711,6 +6074,200 @@ mod tests {
         }
     }
 
+    #[test_case(feature_set_without_require_rent_exempt_split_destination(), Ok(()); "without_require_rent_exempt_split_destination")]
+    #[test_case(feature_set_all_enabled(), Err(InstructionError::InsufficientFunds); "all_enabled")]
+    fn test_split_require_rent_exempt_destination(
+        feature_set: Arc<FeatureSet>,
+        expected_result: Result<(), InstructionError>,
+    ) {
+        let rent = Rent::default();
+        let rent_exempt_reserve = rent.minimum_balance(StakeStateV2::size_of());
+        let stake_history = StakeHistory::default();
+        let current_epoch = 100;
+        let clock = Clock {
+            epoch: current_epoch,
+            ..Clock::default()
+        };
+        let minimum_delegation = crate::get_minimum_delegation(&feature_set);
+        let delegation_amount = 3 * minimum_delegation;
+        let source_lamports = rent_exempt_reserve + delegation_amount;
+        let source_address = solana_sdk::pubkey::new_rand();
+        let destination_address = solana_sdk::pubkey::new_rand();
```

Comment:
```suggestion
        let source_address = solana_sdk::pubkey::new_unique();
        let destination_address = solana_sdk::pubkey::new_unique();
```

for the sake of killing off the old evil

Reply:
There's a whole bunch of this in the other tests in this module. I'll fix the rest in a separate PR